src/WebsocketClient.ts

```diff
@@ -46,44 +47,45 @@ export const PUBLIC_WS_KEYS: WsKey[] = [
 type WsTopic = string;
 
 export class WebsocketClient extends BaseWebsocketClient<WsKey> {
-  private RESTClientCache: Record<
-    WsMarket,
-    SpotClient | FuturesClient | undefined
-  > = {
+  RESTClientCache: Record<WsMarket, SpotClient | FuturesClient | undefined> = {
```

Comment:
You can make these private, since they're not supposed to be accessible (if someone wants them they should make their own instance, shouldn't use the one in the WS client.
```suggestion
  private RESTClientCache: Record<WsMarket, SpotClient | FuturesClient | undefined> = {
```

Reply:
I see the reasoning behind making these private, but I need access to them in my own WS instance because I pass a custom Spot class with specific methods for handling timestamps. Restricting access would limit flexibility for contributors with similar use cases.

Reply:
Thanks for your response!

To clarify, in my custom Spot class, I override the getSignTimestampMs method to implement specific timestamp handling logic. This is important for my use case because I need precise control over how timestamps are generated and signed when interacting with the Kucoin API.

In my KucoinWebsocketClient, I fetch the REST client using getRESTClient(wsKey), and I want to ensure that my custom Spot class—along with its overridden getSignTimestampMs method—is accessible in the WebSocket implementation.

Reply:
Interesting, thanks for clarifying! I did wonder if anyone would start using it. Out of scope for this PR, but for a general planning idea, would it help to be able to inject a custom getSignTimestampMs method into the constructor, similar to how I've started to allow injecting a custom sign message function method?

Example here:
https://github.com/tiagosiebler/bybit-api/blob/wsapi/examples/fasterHmacSign.ts#L64-L68

In the meantime, I'm still cautious how much I expose the embedded REST client in the WS client. Once it's public for a while, I'm essentially in a contract to keep that public and unchanged or risk breaking changes for anyone who might have started using it. That restricts how much I can change/improve over time without risking breaking changes, thus the hesitation here.

To give you what you need in the meantime, how about making the getter and the property both private, but exposing a setRESTClient(wsKey, restClient) that allows you to write your own REST client instance into the object store? The getter only writes to the store if no instance exists yet, so this should still give you the control you need - in a write-only way (reducing the risk of others building around the internally instanced REST client). Hope my explanation makes sense but let me know your thoughts. Thanks!

Reply:
I understand the hesitation in exposing the embedded REST client too much, as it could limit future improvements. The setRESTClient(wsKey, restClient) approach sounds like a reasonable compromise—it would let me inject my custom Spot instance while keeping things encapsulated.

That said, I just want to clarify: if I set my own REST client via setRESTClient, will the WebSocket client always use that instance, or is there any scenario where it might create a new one internally?

Also, regarding the idea of injecting a custom getSignTimestampMs method into the constructor, I think that would be a great long-term solution. It aligns well with the existing custom sign function injection and would provide a cleaner way to handle timestamp customization without needing to override the entire REST client.